src/Uno.UWP/System/Power/PowerManager.iOS.cs

```diff
@@ -17,17 +17,17 @@ public partial class PowerManager
 		private static NSObject _batteryStateChangeSubscription;
 		private static NSObject _powerStateChangeSubscription;
 
-		private static readonly UIDevice _device;
-		private static readonly bool _isSimulator;
+		private static readonly UIDevice _device = UIDevice.CurrentDevice;
+		private static readonly bool _isSimulator = InitializeIsSimulator();
 
-		static PowerManager()
+		private static bool InitializeIsSimulator()
 		{
-			_device = UIDevice.CurrentDevice;
-			_isSimulator = _device.Model?
+			var isSimulator = _device.Model?
 				               .Contains(
 					               DeviceModelSimulator,
 					               StringComparison.InvariantCultureIgnoreCase) == true;
-			_device.BatteryMonitoringEnabled = !_isSimulator;
+			_device.BatteryMonitoringEnabled = !isSimulator;
+			return isSimulator;
```

Comment:
This fees like it's assuming with the order of assignation of the `_device` static field and could hide subtle bugs.

Reply:
The order of execution matches the order of declarations in source code. Do you want me to add a comment stating that order is important, or revert the ones you commented on?